src/components/data/DataTable.js

```diff
@@ -0,0 +1,151 @@
+import { downcastRef } from '../../util/typing';
+import { useArrowKeyNavigation } from '../../hooks/keyboard-navigation';
+import { useElementRef } from '../../hooks/use-element-ref';
+
+import { SpinnerSpokesIcon } from '../icons';
+import Table from './Table';
+import TableHead from './TableHead';
+import TableBody from './TableBody';
+import TableRow from './TableRow';
+import TableCell from './TableCell';
+/**
+ * @typedef {import('../../types').CompositeProps} CommonProps
+ * @typedef {Omit<import('preact').JSX.HTMLAttributes<HTMLElement>, 'size'|'rows'|'role'|'loading'>} HTMLAttributes
+ * @typedef {import('./TableContext').TableInfo} TableInfo
+ * @typedef {import('preact').ComponentChildren} Children
+ *
+ * @typedef Column
+ * @prop {string} field
+ * @prop {string} label
+ * @prop {string} [classes]
+ *
+ * @typedef {Record<string, any>} Row
+ *
+ * @typedef {( row:Row, field:string) => Children|null} RenderItem
+ * @typedef {( row: Row ) => void } RowCallback
+ *
+ * @typedef DataTableProps
+ * @prop {Column[]} columns
+ * @prop {Row[]} rows
+ * @prop {string} title
+ * @prop {Row|null} [selectedRow]
+ * @prop {boolean} [loading=false]
+ * @prop {RowCallback} [onConfirmRow]
+ * @prop {RowCallback} [onSelectRow]
+ * @prop {RenderItem} [renderItem] - Callback to render an individual table cell
+ * @prop {Children} [emptyMessage] - Content to render if rows is empty (and not
+ *   in a loading state)
+ */
+
+/**
+ * An interactive table of rows and columns with a sticky header.
+ *
+ * @param {CommonProps & DataTableProps & HTMLAttributes} props
+ */
+const DataTableNext = function DataTable({
+  children,
+  elementRef,
+
+  columns,
+  rows,
+  title,
+  selectedRow,
+  loading = false,
+  renderItem,
+  onSelectRow,
+  onConfirmRow,
+  emptyMessage,
+
+  ...htmlAttributes
+}) {
+  const tableRef = useElementRef(elementRef);
+
+  useArrowKeyNavigation(tableRef, {
+    selector: 'tbody tr',
+    horizontal: true,
+    vertical: true,
+  });
```

Comment:
At this time, because we can't use hooks conditionally, _all_ `DataTable`s get keyboard navigation. We may want to be more selective. Any thoughts about that welcome.